lib/ssl/src/tls_connection.erl

```diff
@@ -248,27 +248,27 @@ hello(internal, #server_hello{} = Hello,
              connection_env = CEnv,
 	     static_env = #static_env{role = client},
              handshake_env = #handshake_env{
-                                ocsp_stapling_state = OcspState0,
+                                stapling_state = StaplingState0,
                                 renegotiation = {Renegotiation, _}} = HsEnv,
              session = #session{session_id = OldId},
 	     ssl_options = SslOptions} = State) ->
     try
         case tls_handshake:hello(Hello, SslOptions, ConnectionStates0, Renegotiation, OldId) of
             %% Legacy TLS 1.2 and older
-            {Version, NewId, ConnectionStates, ProtoExt, Protocol, OcspState} ->
+            {Version, NewId, ConnectionStates, ProtoExt, Protocol, StaplingState} ->
                 tls_dtls_connection:handle_session(
                   Hello, Version, NewId, ConnectionStates, ProtoExt, Protocol,
                   State#state{
                     handshake_env =
                         HsEnv#handshake_env{
-                          ocsp_stapling_state = maps:merge(OcspState0,OcspState)}});
+                          stapling_state = maps:merge(StaplingState0,StaplingState)}});
             %% TLS 1.3
-            {next_state, wait_sh, SelectedVersion, OcspState} ->
+            {next_state, wait_sh, SelectedVersion, StaplingState} ->
                 %% Continue in TLS 1.3 'wait_sh' state
                 {next_state, wait_sh,
                  State#state{handshake_env =
-                                 HsEnv#handshake_env{ocsp_stapling_state =
-                                                         maps:merge(OcspState0, OcspState)},
+                                 HsEnv#handshake_env{stapling_state =
```

Comment:
When running `-suite openssl_stapling_SUITE -group 'tlsv1.3' -case staple_with_nonce` - this section does not get executed. How it works for TLS1.3 ??

Reply:
Maybe this is how it should actually work as described in https://datatracker.ietf.org/doc/html/rfc8446#section-4.4.2.1.

Reply:
This stapling code seems dead - maybe remove.

Reply:
for TLS 1.3 status_request extension (server response completing extension negotiation) is checked `tls_handshake_1_3:split_cert_entries` not in ServerHello.
merging StaplingState here is not needed I guess.

@IngelaAndin

Reply:
I guess you are right, try removing it and run the tests. No it will not be covered by refactor other than the code being placed in an other file called client something.

Reply:
> I guess you are right, try removing it and run the tests. No it will not be covered by refactor other than the code being placed in an other file called client something.

Do you mean `maps:merge(StaplingState0, StaplingState)` part or whole TLS1.3 clause (I think coverage report marks it as used, but I don't understand what scenario that could be)

Reply:
Clarification it will then change to the TLS-1.3 state machine. 

```
 [{change_callback_module, tls_client_connection_1_3},
                  {next_event, internal, Hello}]}
```

Reply:
I think I will leave this code as is - leave the merge.
For TLS1.3, in this clause we will only initialized StaplingState to `#{configured => true, status => not_negotiated}`